File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int mini=INT_MAX;
        int pro=0;
        for(int i=0;i<n;i++){
            mini=min(mini,prices[i]);
            pro=max(prices[i]-mini,pro);
            
        }
        return pro;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxi = 0
        buy = 0

        for sell in range(1, len(prices)):
            if prices[buy] < prices[sell]:
                profit = prices[sell] - prices[buy]
                if profit > maxi:
                    maxi = profit
            else:
                buy = sell

        return maxi
        
# TC - O(N)
# SC - O(1)

# Pattern - Two Pointer